types/libnpmpublish/index.d.ts

```diff
@@ -0,0 +1,39 @@
+// Type definitions for libnpmpublish 4.0
+// Project: https://npmjs.com/package/libnpmpublish
+// Definitions by: nicolas377 <https://github.com/nicolas377>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import fetch = require("npm-registry-fetch");
+import nodefetch = require("node-fetch");
+import Response = nodefetch.Response;
+
+/**
+ * Sends the package represented by the manifest and tarData to the configured registry.
+ * @param manifest the parsed package.json for the package being published (which can also be the manifest pulled from a packument, a git repo, tarball, etc.)
+ * @param tarballData a Buffer of the tarball being published.
+ * @param options (optional) a object containing the options
+ * @returns a Promise that resolves to a Response
+ * @example // note that pacote.manifest() and pacote.tarball() can also take
+ * // any spec that npm can install.  a folder shown here, since that's
+ * // far and away the most common use case.
+ * const path = '/a/path/to/your/source/code'
+ * const pacote = require('pacote') // see: http://npm.im/pacote
+ * const manifest = await pacote.manifest(path)
+ * const tarData = await pacote.tarball(path)
+ * await libpub.publish(manifest, tarData, { npmVersion: 'my-pub-script@1.0.2', token: 'my-auth-token-here' }, opts)
+ * // Package has been published to the npm registry.
+ */
+export function publish(manifest: object, tarballData: Buffer, options?: fetch.Options): Promise<Response>;
```

Comment:
Should we lock in the manifest object to match pacote's manifest type? - https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pacote/index.d.ts#L80

Reply:
Just committed with a fix to that